draft-lenders-dns-cbor.md

```diff
@@ -627,12 +555,100 @@ dns-response = [
 
-# Further Compression with CBOR-packed {#sec:cbor-packed}
+# Compression with CBOR-packed {#sec:cbor-packed}
 
 If both DNS server and client support CBOR-packed {{-cbor-packed}}, it MAY be used for further
 compression in DNS responses.
 Especially IPv6 addresses, e.g., in AAAA resource records can benefit from straight referencing to
-compress common address prefixes.
+compress common address prefixes.[^6]{: mlenders}
+
+[^6]: Needs rewording.
+
+## Name Compression {#sec:name-compression}
+
+~~~ cddl
+Text-String-Suffix-Sequence-Packed-CBOR = #6.28259(rump)
+~~~
+
+For name compression, a new packing table setup tag TBD28259 ('n' and 'c' in ASCII) for CBOR-packed {{-cbor-packed}} is defined.
+It provides an implicit text string suffix sequence table for shared items _V_ which is prepended to the existing table for shared items that apply to the entire tag TBD28259 (by default empty table).
+The this implicit (i.e. not explicitly represented) table _V_ is constructed as follows:
+Any coherent sequence of text strings encountered within the rump of tag TBD28259, as well as any of its non-empty suffixes, are added to the table as an array in depth-first order.
+Text string sequences within any tables for shared items or argument items within the rump MUST not be added to _V_.
+If a sequence for which an array is already in _V_ is encountered, a shared item reference _i_ to that array in V replaces this sequence.
+This shared item reference _i_ means: take the sequence from the array at _V_\[_i_\] and put it into the surrounding array in place of _i_.
+The resulting rump should look like referencing the _i_-th string (depth first) in the sequence.
+
+The "application/dns+cbor" media type comes with an optional parameter "packed".
+If it is not provided, it is assumed to be 0.
+With packed=0, any CBOR object `obj` marked by the "application/dns+cbor" media type MUST explicitly be understood as `TBD28259(obj)`, unless it is already `obj` itself is already tagged explicitly with TBD28259 as a whole.
+
+### Example
+
+Take the following content (note that this is intentionally not legal "application/dns+cbor" to illustrate generality).
+
+~~~ edn
+[
+  "www", "example", "org",
+  ["svc", "www", "example", "org"],
+  "org", "example", "org", 42,
+  "svc", "www", "example", "org", 42
+]
+~~~
+{: #fig:name-compression-example-unpacked title="Unpacked example for implicit text string suffix sequence compression."}
+
+This would generate the following virtual table _V_.
+
+~~~ edn
+[
+    ["www", "example", "org"],
+    ["example", "org"],
+    ["org"],
+    ["svc", "www", "example", "org"],
+    ["org", "example", "org"]
+]
+~~~
+{: #fig:name-compression-example-table title="Implicit table of shared items for the example."}
+
+Note that the sequence "org", "example", "org" is added at index 4 with leading "org", instead of referencing index 2 + index 1, as it is its own distinct suffix sequence.
```

Comment:
```suggestion
Note that the sequence "org", "example", "org" is added at index 4 with leading "org", instead of referencing index 2 + index 1 (`simple(2), simple(1)`), as it is its own distinct suffix sequence.
```